testsetups/vrfv2.go

```diff
@@ -0,0 +1,169 @@
+	ChainlinkNodeFunding *big.Float    // Amount of ETH to fund each chainlink node with
+	StopTestOnError      bool          // Do we want the test to stop after any error or just continue on
+
+	RequestsPerMinute int                   // Number of requests for randomness per minute
+	TestFunc          VRFV2SoakTestTestFunc // The function that makes the request and validations wanted
+}
+
+// VRFV2SoakTestJobInfo defines a jobs into and proving key info
+type VRFV2SoakTestJobInfo struct {
+	Job            *client.Job
+	ProvingKey     [2]*big.Int
+	ProvingKeyHash [32]byte
+}
+
+// NewVRFV2SoakTest creates a new vrfv2 soak test to setup and run
+func NewVRFV2SoakTest(inputs *VRFV2SoakTestInputs) *VRFV2SoakTest {
+	return &VRFV2SoakTest{
+		Inputs: inputs,
+		TestReporter: testreporters.VRFV2SoakTestReporter{
+			Reports: make(map[string]*testreporters.VRFV2SoakTestReport),
+		},
+	}
+}
+
+// Setup sets up the test environment
+func (t *VRFV2SoakTest) Setup(env *environment.Environment, isLocal bool) {
+	t.ensureInputValues()
+	t.env = env
+	var err error
+	var networkRegistry *blockchain.NetworkRegistry
+
+	// Make connections to soak test resources
+	if isLocal {
+		err = env.ConnectAll()
+		Expect(err).ShouldNot(HaveOccurred())
+		networkRegistry = blockchain.NewDefaultNetworkRegistry()
+		t.ChainlinkNodes, err = client.ConnectChainlinkNodes(env)
+		Expect(err).ShouldNot(HaveOccurred(), "Connecting to chainlink nodes shouldn't fail")
+		t.mockServer, err = client.ConnectMockServer(env)
+		Expect(err).ShouldNot(HaveOccurred(), "Creating mockserver clients shouldn't fail")
+	} else {
+		networkRegistry = blockchain.NewSoakNetworkRegistry()
+		t.ChainlinkNodes, err = client.ConnectChainlinkNodesSoak(env)
+		Expect(err).ShouldNot(HaveOccurred(), "Connecting to chainlink nodes shouldn't fail")
+		t.mockServer, err = client.ConnectMockServerSoak(env)
+		Expect(err).ShouldNot(HaveOccurred(), "Creating mockserver clients shouldn't fail")
+	}
+
+	t.Networks, err = networkRegistry.GetNetworks(env)
+	Expect(err).ShouldNot(HaveOccurred(), "Connecting to blockchain nodes shouldn't fail")
+	t.DefaultNetwork = t.Networks.Default
+	t.DefaultNetwork.ParallelTransactions(true)
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+// Run starts the VRFV2 soak test
+func (t *VRFV2SoakTest) Run() {
+	log.Info().
+		Str("Test Duration", t.Inputs.TestDuration.Truncate(time.Second).String()).
+		Int("Max number of requests per second wanted", t.Inputs.RequestsPerMinute).
+		Msg("Starting VRFV2 Soak Test")
+
+	// set the requests to only run for a certain amount of time
+	testContext, testCancel := context.WithTimeout(context.Background(), t.Inputs.TestDuration)
+	defer testCancel()
+
+	t.NumberOfRequests = 0
+
+	// variables dealing with how often to tick and how to stop the ticker
+	stop := false
+	startTime := time.Now()
+	ticker := time.NewTicker(time.Minute / time.Duration(t.Inputs.RequestsPerMinute))
```

Comment:
This might be easier with a context? [I did OCR with one](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/testsetups/ocr.go#L127) and was pleased, but it might not work for your specific case. Maybe worth a look.

Reply:
Your implementation doesn't tick on the same interval every time. It is ticking every (time it takes to make all the calls + TimeBetweenRounds) which is too limiting for some of the higher requests per minute the vrf team want.